From 5f38727a19f61f303e615fab469c8ff9a4bab63d Mon Sep 17 00:00:00 2001
From: Jithu Joseph <jithu.joseph@intel.com>
Date: Thu, 24 Feb 2022 15:03:37 -0800
Subject: [PATCH 1068/1418] Revert "x86/microcode/intel: expose
 collect_cpu_info_early() for SAF"

This reverts commit 5101195b76d837c2d88f6ae7542e8093f82c6a39.

Signed-off-by: Jithu Joseph <jithu.joseph@intel.com>
---
 arch/x86/include/asm/microcode_intel.h | 6 ------
 arch/x86/kernel/cpu/microcode/intel.c  | 8 +++-----
 2 files changed, 3 insertions(+), 11 deletions(-)

diff --git a/arch/x86/include/asm/microcode_intel.h b/arch/x86/include/asm/microcode_intel.h
index f6beb5edce8c..a342234a0a5b 100644
--- a/arch/x86/include/asm/microcode_intel.h
+++ b/arch/x86/include/asm/microcode_intel.h
@@ -74,9 +74,6 @@ extern void load_ucode_intel_ap(void);
 extern void show_ucode_info_early(void);
 extern int __init save_microcode_in_initrd_intel(void);
 void reload_ucode_intel(void);
-int collect_cpu_info_early(struct ucode_cpu_info *uci);
-bool cpu_signatures_match(unsigned int s1, unsigned int p1,
-			  unsigned int s2, unsigned int p2);
 int update_cpusvn_intel(void);
 #else
 static inline __init void load_ucode_intel_bsp(void) {}
@@ -84,9 +81,6 @@ static inline void load_ucode_intel_ap(void) {}
 static inline void show_ucode_info_early(void) {}
 static inline int __init save_microcode_in_initrd_intel(void) { return -EINVAL; }
 static inline void reload_ucode_intel(void) {}
-static inline void collect_cpu_info_early(struct ucode_cpu_info *uci) {}
-static inline void cpu_signatures_match(unsigned int s1, unsigned int p1,
-					unsigned int s2, unsigned int p2) {}
 static inline int update_cpusvn_intel(void) { return -1; }
 #endif
 
diff --git a/arch/x86/kernel/cpu/microcode/intel.c b/arch/x86/kernel/cpu/microcode/intel.c
index b2982c1b0b84..5e9dcd645913 100644
--- a/arch/x86/kernel/cpu/microcode/intel.c
+++ b/arch/x86/kernel/cpu/microcode/intel.c
@@ -48,8 +48,8 @@ static struct microcode_intel *intel_ucode_patch;
 static int llc_size_per_core;
 extern bool ucode_rollback;
 
-bool cpu_signatures_match(unsigned int s1, unsigned int p1,
-			  unsigned int s2, unsigned int p2)
+static inline bool cpu_signatures_match(unsigned int s1, unsigned int p1,
+					unsigned int s2, unsigned int p2)
 {
 	if (s1 != s2)
 		return false;
@@ -61,7 +61,6 @@ bool cpu_signatures_match(unsigned int s1, unsigned int p1,
 	/* ... or they intersect. */
 	return p1 & p2;
 }
-EXPORT_SYMBOL_GPL(cpu_signatures_match);
 
 /*
  * Returns 1 if update has been found, 0 otherwise.
@@ -346,7 +345,7 @@ scan_microcode(void *data, size_t size, struct ucode_cpu_info *uci, bool save)
 	return patch;
 }
 
-int collect_cpu_info_early(struct ucode_cpu_info *uci)
+static int collect_cpu_info_early(struct ucode_cpu_info *uci)
 {
 	unsigned int val[2];
 	unsigned int family, model;
@@ -376,7 +375,6 @@ int collect_cpu_info_early(struct ucode_cpu_info *uci)
 
 	return 0;
 }
-EXPORT_SYMBOL_GPL(collect_cpu_info_early);
 
 static void show_saved_mc(void)
 {
-- 
2.31.1

